source/core.py

```diff
@@ -584,9 +584,14 @@ def run(self):
 		log.debug("initializing updateCheck")
 		updateCheck.initialize()
 	log.info("NVDA initialized")
+
 	# Queue the firing of the postNVDAStartup notification.
 	# This is queued so that it will run from within the core loop,
 	# and initial focus has been reported.
+	def _postNvdaStartupLogMessage():
+		log.debug("postNvdaStartup event fired")
+
+	postNvdaStartup.register(_postNvdaStartupLogMessage)
 	queueHandler.queueFunction(queueHandler.eventQueue, postNvdaStartup.notify)

```

Comment:
One thing I hadn't considered when we talked about this, registering will mean the order of the log message isn't guaranteed. I think it would probably be better if the log message was known to be before the action is fired, given the cause effect relationship, explicit ordering is much easier to reason about.

I'd prefer:
```suggestion
	def _doPostNvdaStartupAction():
		log.debug("Notify of postNvdaStartup action")
		postNvdaStartup.notify()

	queueHandler.queueFunction(queueHandler.eventQueue, _doPostNvdaStartupAction)

```